README.md

```diff
@@ -144,6 +144,31 @@ kubectl apply -f elastic-agent-${ELASTIC_AGENT_MODE}-kubernetes.yaml
 kubectl -n kube-system get pods -l app=elastic-agent
 ```
 
+## Testing on Elastic Cloud
+
+Elastic employees can create an Elastic Cloud deployment with a locally
+built Elastic Agent, by pushing images to an internal Docker repository. The images will be 
+based on the SNAPSHOT images with the version defined in `version/version.go`.
+
+Prerequisite to running following commands is having `terraform` installed and running `terraform init` from within `testing/environments/cloud`.
+
+Running `mage cloud:image` in this directory or `make build_elastic_agent_docker_image` in `testing/environments/cloud` will build and push the images. 
+Running `mage cloud:push` in this directory or `make push_elastic_agent_docker_image` in `testing/environments/cloud` will publish built docker image to CI docker repository.
+
+Once docker images are published you can run `EC_API_KEY=your_api_key make apply` from `testing/environments/cloud` directory to deploy them to Elastic Cloud. 
```

Comment:
You should mention the need to update the `docker_image.auto.tfvars` file here, because running `mage cloud:push` doesn't do anything helpful unless you save the tag it creates and update the Terraform plan with it.

I actually think we should mention `make deploy` local first since it does what most people will actually want by templating the version into docker tfvars file automatically.